synapseclient/models/file.py

```diff
@@ -1013,13 +1021,15 @@ async def get_async(
             if_collision=self.if_collision,
             limit_search=self.synapse_container_limit or self.parent_id,
             download_file=self.download_file,
-            download_location=self.download_location,
+            download_location=os.path.dirname(self.path)
+            if self.path and os.path.isfile(self.path)
+            else self.path,
```

Comment:
Nice! I'm thinking about this a little bit more:

What happens if you store a file with `path=/user/bfauble/folder/file.txt` for example, and I try to download it. Does my machine automatically make a folder called `bfauble` and a subfolder called `folder` to store my file?

Another thought, but related: This looks to me that if a folder you're downloading has `path` in the filehandle, it will replace synapse cache as the new default, which is new behavior and should be heavily documented everywhere

Reply:
> What happens if you store a file with path=/user/bfauble/folder/file.txt for example, and I try to download it. Does my machine automatically make a folder called bfauble and a subfolder called folder to store my file?

No. The file will be downloded based on the `path` that you specify, or the `.synapseCache` if no path is specified.

> Another thought, but related: This looks to me that if a file you're downloading has path in the filehandle, path will replace synapse cache as the new default, which is new behavior and should be heavily documented everywhere


The is the current behavior of the client through the current Synapse class.

> Another thought: Have you tested this for a case where you're downloading a Folder, and 2 Files within the folder have different paths?

This isn't a scenario that would happen I don't think. You can use the Folder.sync_from_synapse function for sure, but that function takes in a `path` argument, and all files/folders under it will be downloaded to the respective location(s) based on how they're organized in Synapse.

Reply:
> No. The file will be downloded based on the path that you specify, or the .synapseCache if no path is specified.

You're right, I see that now. `self.path` doesn't get assigned `path` from an entity, or anything like that, if it's not declared.

Thanks!